### ISSUE TITLE
How to Execute Nested Queries?

### ISSUE THREAD

**Issue by rintuzi123** [brody2consult/cordova-sqlite-storage#435]
Hi dev. I'm using Cordova-sqlite-storage in my Ionic App and it really works great. But I want to know is there any way to execute nested queries in this plugin?

`var query = "SELECT * FROM Company";
$cordovaSQLite.execute(db, query, []).then(function (res) {
  if (res.rows.length > 0) {
                        for (var i = 0; i < res.rows.length; i++) {
                                $cordovaSQLite.execute(db, "SELECT * FROM Employee WHERE CompanyCode = ?", [res.rows.item(i).CompanyCode]).then(function (employee) {
alert(employee.rows.item(0).EmployeeName);
                                }, function (e) {
                                    jsShowPageMessage('Error - ' + JSON.stringify(e), 'error');
                                });
                            }
                        }
            }, function (err) {
                jsShowPageMessage('Error - ' + JSON.stringify(err), 'error');
            });`

Here the problem is "As the execute calls are asynchronous the outer execute process is not waiting for the inner execute process to complete". Plz let me know how to achieve my goal.

**Comment by brody2consult** [comment]
Your code is very hard to follow and I cannot understand what you are trying to achieve.

The asynchronous transaction API allows you to execute multiple queries, and to add more queries in executeSql callbacks over and over, within a single transaction. The asynchronous transaction API is almost the same as the Web SQL API (with the known limitations and deviations documented in README.md). The asynchronous Web SQL transaction API is very well documented.

I will add add some tips when I get a chance.
